nanocoat/lib/scritchui/scritchPencilRasterShapes.c

```diff
@@ -17,7 +17,399 @@
 #include "sjme/debug.h"
 #include "sjme/fixed.h"
 
-sjme_errorCode sjme_scritchpen_core_drawRect(
+sjme_errorCode sjme_scritchpen_corePrim_drawArc(
+	sjme_attrInNotNull sjme_scritchui_pencil g,
+	sjme_attrInValue sjme_jint x,
+	sjme_attrInValue sjme_jint y,
+	sjme_attrInPositive sjme_jint w,
+	sjme_attrInPositive sjme_jint h,
+	sjme_attrInValue sjme_jint startAngle,
+	sjme_attrInValue sjme_jint arcAngle)
+{
+	sjme_errorCode error;
```

Comment:
Guard function with


```c
#if defined(SJME_CONFIG_HAS_FLOAT_HARD)
	...
#else
	sjme_todo("Impl?");
	return sjme_error_notImplemented(0);
#endif
```

Reply:
Since this will already guard all of the float usage in the function, and is the same guard as for sjme_jfloat itself i'd rather not use sjme_jfloat explicitly since it is a union, which ended up making parts of fillArc and drawArc quite cumbersome to read due to all the manipulation.